lib/ansible/modules/cloud/amazon/ssm_send_command.py

```diff
@@ -0,0 +1,211 @@
+    type: str
+    sample: Success
+'''
+
+from ansible.module_utils.basic import AnsibleModule, BOOLEANS
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+import traceback
+from time import sleep
+
+try:
+    import boto3
+    import botocore
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name                 = dict(required=True),
+        wait                 = dict(choices=BOOLEANS, default=True, type='bool'),
+        comment              = dict(),
+        instanceIds          = dict(required=True, type='list'),
+        parameters           = dict(default={}, type='dict')
+    ))
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False
+    )
+
+    document_name = module.params.get('name')  # Needs to be an existing SSM document name
+    comment = module.params.get('comment')
+    await_return = module.params.get('wait')
+    instance_ids = module.params.get('instanceIds')
+    parameters = module.params.get('parameters')
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='Python module "boto3" is missing, please install it')
+
+    if not (document_name and instance_ids):
+        module.fail_json(msg="Must provide SSM document name and at least one instance id.")
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=HAS_BOTO3)
+    if not region:
+        module.fail_json(msg="The AWS region must be specified as an "
+                         "environment variable or in the AWS credentials "
+                         "profile.")
+
+    try:
+        client = boto3_conn(module, conn_type='client', resource='ssm',
+                            region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except (botocore.exceptions.ClientError, botocore.exceptions.ValidationError) as e:
+        module.fail_json(msg="Failure connecting boto3 to AWS", exception=traceback.format_exc(e))
+
+    invoke_params = {}
+
+    if document_name:
+        invoke_params['DocumentName'] = document_name
+    if comment:
+        invoke_params['Comment'] = comment
+    if instance_ids:
+        invoke_params['InstanceIds'] = instance_ids
+    if parameters:
+        invoke_params['Parameters'] = parameters
+
+    try:
+        response = client.send_command(**invoke_params)
+    except botocore.exceptions.ClientError as ce:
+        if ce.response['Error']['Code'] == 'ResourceNotFoundException':
+            module.fail_json(msg="Could not find the SSM doc to execute. Make sure "
+                             "the document name is correct and your profile has "
+                             "permissions to execute SSM.",
+                             exception=traceback.format_exc(ce))
+        module.fail_json(msg="Client-side error when invoking SSM, check inputs and specific error",
```

Comment:
I would follow the guidelines here for boto3 exception handling https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-2